src/utils/git.ts

```diff
@@ -13,37 +13,41 @@ const excludeBigFilesFromDiff = ['*-lock.*', '*.lock'].map(
   (file) => `:(exclude)${file}`
 );
 
-export interface StagedDiff {
-  files: string[];
-  diff: string;
-}
-
-export const getStagedGitDiff = async (
-  isStageAllFlag = false
-): Promise<StagedDiff | null> => {
-  if (isStageAllFlag) {
-    const stageAllSpinner = spinner();
-    stageAllSpinner.start('Staging all changes');
-    await execa('git', ['add', '.']);
-    stageAllSpinner.stop('Done');
-  }
-
-  const diffStaged = ['diff', '--staged'];
+export const getStagedFiles = async (): Promise<string[]> => {
   const { stdout: files } = await execa('git', [
-    ...diffStaged,
+    'diff',
     '--name-only',
-    ...excludeBigFilesFromDiff
```

Comment:
why do you remove this? package-lock.json is gonna be counted like a change, and it's too big for openAI api, splitting it in chunks looks like overengineering and waste of tokens sent to the API. I'd provide an error message if only `excluded` files are staged like: "staged files are .lock files which are excluded by default as it's  too big, commit it yourself, don't waste your api tokens". WDYT?

Reply:
@vladickweb bro :)

Reply:
Sorry, I replied to you but for some reason, it didn't send 😅

The function `getStagedFiles` indeed returns all the file names that have been modified but not their diffs. That's why we don't exclude "lock" files there, but we do exclude them in the `getDiff` function.

Regarding adding or updating a new dependency, it should be reflected in the package.json file and thus, it is implied that the lock file has also been changed. Can you think of any ideas to improve it? :)

Reply:
i mean, if you don't exclude .lock files from `getStagedFiles` and do exclude .lock files from `getDiff` this happens:

1. user runs `oc`
2. `getStagedFiles` sees .lock and goes further (no error, we have a staged file)
3. `getDiff` get's nothing as .lock is excluded
4. error :)

so i suggest excluding .lock in both functions and `if only excluded files are staged` show an error message: `staged files are .lock files which are excluded by default as it's too big, commit it yourself, don't waste your api tokens`

Reply:
done bro, check it! ❤️